dali/pipeline/executor/executor.cc

```diff
@@ -299,6 +306,24 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunHelper(OpNode &op_node, Workspac
   const auto &schema = spec.GetSchema();
   SmallVector<int, 16> empty_layout_in_idxs;
 
+  cudaStream_t prev_stage_stream = ws.has_stream() && ws.stream() == gpu_op_stream_
```

Comment:
What is the previous stream for the CPU stage?

Reply:
It will be gpu_op_stream_ - same as for mixed, since in pure CPU stage there's no usage of streams.